File: Find_Depth.c
int FindDepth(struct Node* root) {
    if (root == NULL)
        return 0;

    int leftDepth = FindDepth(root->left);
    int rightDepth = FindDepth(root->right);

    return (leftDepth > rightDepth) ? (leftDepth + 1) : (rightDepth + 1);
}

File: Find_Depth.java
public int findDepth(Node root) {
    if (root == null)
        return 0;

    int leftDepth = findDepth(root.left);
    int rightDepth = findDepth(root.right);

    return (leftDepth > rightDepth) ? (leftDepth + 1) : (rightDepth + 1);
}
